ai/agents.py

```diff
@@ -131,9 +136,11 @@ def on_captured(self) -> None:
         """ Called once when the agent is captured """
         self.log('I\'ve been captured... :(')
 
-    def on_noise(self, noise: world.NoiseEvent) -> None:
+    def on_noise(self, noises: List['world.NoiseEvent']) -> None:
         """ Noise handler, will be called before `on_tick` """
-        ...
+        angle = super().on_noise(noises);
```

Comment:
Same comment here as a bit further down, calling `super` here should never happen.
1) It's quite confusing
2) It's an abstract method, so you can safely assume that the parent implementation is empty

Reply:
In that case I would have to write the code for each implementation even though this is a behaviour that all agents show.